src/System.Management.Automation/engine/lang/parserutils.cs

```diff
@@ -963,14 +962,29 @@ internal static object ReplaceOperator(ExecutionContext context, IScriptExtent e
         /// <returns>The result of the regex.Replace operation</returns>
         private static object ReplaceOperatorImpl(ExecutionContext context, string input, Regex regex, object substitute)
         {
-            MatchEvaluator matchEvaluator = null;
-            if (!(substitute is string) && LanguagePrimitives.TryConvertTo(substitute, out matchEvaluator))
+            switch (substitute)
             {
-                return regex.Replace(input, matchEvaluator);
-            }
+                case ScriptBlock sb:
+                    MatchEvaluator me = match => {
+                        var result = sb.DoInvokeReturnAsIs(
+                            useLocalScope: false, /* Use local scope to be consistent with 'ForEach/Where-Object {}' and 'collection.ForEach{}/Where{}' */
```

Comment:
I don't  understand the comment here - we use `false`.

Reply:
Good catch. It's a typo, should be "use current scope ..."